### PR TITLE
Add least connections load balancing algorithm

### PR DESCRIPTION
**Pull request by NotBad4U**
works to continue on #409

### CODE REVIEW

File: lib/src/network/mod.rs [hunk 1]
```diff
@@ -204,14 +204,14 @@ pub enum SocketType {
   FrontClient
 }
 
-#[derive(Debug,PartialEq,Eq)]
+#[derive(Debug,PartialEq,Eq,Clone)]
```
Comment (Geal) [inline lib/src/network/mod.rs]: why do you need to implement `Clone` on all those structs now? The `Backend` is already inside a `Rc<RefCell<..>>`

Comment (NotBad4U) [reply lib/src/network/mod.rs]: Not a problem for `when there is no`. That was for the unit test "it_should_find_the_backend_with_least_connections" where I clone the backend `Rc::new(RefCell::new(backend_with_least_connection.clone())),`. Do you mind if I do this?

Comment (Geal) [reply lib/src/network/mod.rs]: I'd prefer you wrap it in a Rc in the test too, then clone that instead of adding those traits just for the test

Comment (NotBad4U) [reply lib/src/network/mod.rs]: Ok I'll try

**Comment by NotBad4U** [comment]
I removed all the `clone`

**Pull request merged**